src/PrestaShopBundle/ApiPlatform/Provider/QueryListProvider.php

```diff
@@ -119,6 +114,48 @@ public function provide(Operation $operation, array $uriVariables = [], array $c
             );
         }
 
-        return new PaginationElements($count, $orderBy, $sortOrder, $limit, $offset, $filters, $normalizedQueryResult);
+        return new PaginationElements(
+            $count,
+            $filter->getOrderBy(),
+            $filter->getOrderWay(),
+            $filter->getLimit(),
+            $filter->getOffset(),
+            $filter->getFilters(),
+            $normalizedQueryResult
+        );
+    }
+
+    private function createFilters(array $context, string $filterClass): Filters
+    {
+        $queryParameters = $context['filters'] ?? [];
+        $paginationParameters = [
+            'filters' => $queryParameters['filters'] ?? [],
+            'orderBy' => array_key_exists('orderBy', $queryParameters) ? $queryParameters['orderBy'] : null,
+            'sortOrder' => array_key_exists('sortOrder', $queryParameters) ? $queryParameters['sortOrder'] : 'asc',
+            'offset' => array_key_exists('offset', $queryParameters) ? (int) $queryParameters['offset'] : null,
+            'limit' => array_key_exists('limit', $queryParameters)
+                ? (int) $queryParameters['limit']
+                : self::DEFAULT_PAGINATED_ITEM_LIMIT,
+        ];
+
+        /* remove null parameters from the request so the default filter parameters can be used instead of null values. */
+        foreach ($paginationParameters as $key => $parameter) {
+            if ($parameter === null) {
+                unset($paginationParameters[$key]);
+            }
+        }
+
+        $request = $this->requestStack->getCurrentRequest();
```

Comment:
```suggestion
        $request = $this->requestStack->getMainRequest();
```
I would have used main request, but I'm not sure it really changes much in this case anyway